### ISSUE TITLE
Add support for upgrade warnings for unknown upgrades according to BIP9

### ISSUE THREAD

**Issue by sondreb** [block-core/blockcore#316]
*Warning mechanism
To support upgrade warnings, an extra "unknown upgrade" is tracked, using the "implicit bit" mask = (block.nVersion & ~expectedVersion) != 0. Mask will be non-zero whenever an unexpected bit is set in nVersion. Whenever LOCKED_IN for the unknown upgrade is detected, the software should warn loudly about the upcoming soft fork. It should warn even more loudly after the next retarget period (when the unknown upgrade is in the ACTIVE state).*

https://github.com/bitcoin/bips/blob/master/bip-0009.mediawiki

**Comment by dangershony** [comment]
yes this will be very useful, to who ever implements this it would also be nice to have the link to where this is handled in bitcoin. core

**Comment by sondreb** [comment]
This is how core gives warning:

2021-03-30T10:40:38Z UpdateTip: new best=0000000000000000000a248163424859beef941ef1eb87085a8e3f3ac7eddca5 height=676985 version=0x20800000 log2_work=92.773308 tx=629443904 date='2021-03-30T10:40:14Z' progress=1.000000 cache=161.9MiB(850517txo) warning='66 of last 100 blocks have unexpected version'
